### PR TITLE
Fix spinners playing looping sound too long in the editor

### LINKED ISSUE #12119: Spinner spin sound issues when scrolling in the timeline.
When scrolling in the timeline while a spinner is playing, the sound of the spinner continues for longer than intended or even starts at different positions. When you click to a different spot on the timeline while the spinner sound is playing, the sound will continually play until you go back to the spinner. This issue perseveres through all trials and tribulation as seen in the video.

Video: https://streamable.com/v51u09

osu!lazer 2021.320.0

**Logs:**
[database.log](https://github.com/ppy/osu/files/6176787/database.log)
[performance.log](https://github.com/ppy/osu/files/6176788/performance.log)
[runtime.log](https://github.com/ppy/osu/files/6176789/runtime.log)
[updater.log](https://github.com/ppy/osu/files/6176790/updater.log)

### PR DESCRIPTION
**Pull request by peppy**
The `OnComplete` event was never being run due to the transform playing out longer than the spinner's lifetime. I've matched the durations, but also moved the `Stop()` call to what I deem a safer place to run it (I did notice that without this it would still potentially never fire).

Note that this is more noticeable in the editor because of lifetime extension. In gameplay, the returning of a spinner to the pool will clean things up (but in the editor that can take longer, depending on timeline zoom level).

Another thing worth mentioning is that the fade doesn't actually work.  This is due to https://github.com/ppy/osu-framework/pull/4212.

Closes #12119.

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs [hunk 1]
```diff
@@ -136,7 +138,10 @@ private void updateSpinningSample(ValueChangedEvent<bool> tracking)
             }
             else
             {
-                spinningSample?.VolumeTo(0, 300).OnComplete(_ => spinningSample.Stop());
+                if (spinningSample != null)
+                    spinningSample.Volume.Value = 0;
+
+                spinningSample?.VolumeTo(0, fade_out_duration);
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs:144]: I'm a bit confused. So it's an instantaneous fade to 0... followed by a transform? Is this right?

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs [hunk 2]
```diff
@@ -173,7 +178,14 @@ protected override void UpdateHitStateTransforms(ArmedState state)
         {
             base.UpdateHitStateTransforms(state);
 
-            this.FadeOut(160).Expire();
+            this.FadeOut(fade_out_duration).OnComplete(_ =>
+            {
+                // looping sample should be stopped here as it is safer than running in the OnComplete
+                // of the volume transition above.
+                spinningSample.Stop();
+            });
```
Comment (bdach) [inline osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs:186]: `UpdateHitStateTransforms()` also handles `ArmedState.Idle`. This logic should probably not execute in that state.

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs (hunk 1 above)
Comment (peppy) [reply osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs:144]: The set to zero was a debug line; unintentional.

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs (hunk 2 above)
Comment (peppy) [reply osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs:186]: I don't think it can cause any harm by doing so, but I guess it's safe to not run.

Comment (peppy) [reply osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs:186]: Actually I'm not sure about this any more. It was previously here and is likely being used to actually decide the lifetime (the `FadeOut` before the `Expire`). I think it's probably fine to leave as is?

**Review by smoogipoo — APPROVED** [review]
I think this is fine as is.

**Pull request merged**